lib/Sema/TypeCheckPattern.cpp

```diff
@@ -738,9 +738,14 @@ static bool validateParameterType(ParamDecl *decl, DeclContext *DC,
                                   TypeChecker &TC) {
   if (auto ty = decl->getTypeLoc().getType())
     return ty->is<ErrorType>();
-  
+
+  // If the element is a variadic parameter, resolve the parameter type as if
+  // it were in non-parameter position, since we want functions to be
+  // @escaping in this case.
```

Comment:
Can you remind me what the array of autoclosures thing was? I can't get variadic autoclosures working at all:

```
Slavas-MBP:~ slava$ cat var.swift 
func foo(fns: @autoclosure () -> Int...) {
  for fn in fns {
    print(fn())
  }
}

foo(1 + 2, 2 + 3, 3 + 4)
Slavas-MBP:~ slava$ xcrun swiftc var.swift
var.swift:7:4: error: missing argument label 'fns:' in call
foo(1 + 2, 2 + 3, 3 + 4)
   ^
    fns: 
var.swift:2:13: error: expression type '[@autoclosure () -> Int]' is ambiguous without more context
  for fn in fns {
            ^~~
```